### PR TITLE
Removes icons from Works dashboard actions dropdown menu

### LINKED ISSUE #5506: Remove icons from the Works dashboard "actions" dropdown menu
### Descriptive summary

The icons in the Works dashboard "actions" dropdown menu are inconsistent with the "actions" dropdown in the Collection dashboard. Given that the icons are not enhancing the accessibility in any way, they should be removed. This is only a UI change as it does not affect backend models.

### Related work

Ref #5490

### PR DESCRIPTION
**Pull request by sdellis**
Fixes #5506 ; refs #5490

@samvera/hyrax-code-reviewers

### CODE REVIEW

File: app/views/hyrax/my/_work_action_menu.html.erb [hunk 1]
```diff
@@ -13,7 +13,6 @@
     <% if can? :edit, document.id %>
       <li role="menuitem" tabindex="-1">
         <%= link_to [main_app, :edit, document] do %>
-          <i class="glyphicon glyphicon-pencil" aria-hidden="true"></i>
           <span> <%= t("hyrax.dashboard.my.action.edit_work") %> </span>
```
Comment (hackartisan) [inline app/views/hyrax/my/_work_action_menu.html.erb:16]: Do we still need the spans once the icons are gone?

**Review by hackartisan — CHANGES_REQUESTED** [review]

**Review by elrayle — DISMISSED** [review]
Looks good to me. 

I noticed that Works use "Edit Work" and Collections use "Edit collection".  A follow-on PR to make those consistent would be good.

**Review by hackartisan — APPROVED** [review]

**Comment by sdellis** [comment]
Screen Shot 2022-03-11 at 11 44 44 AM
Screen Shot 2022-03-11 at 11 41 11 AM

File: app/views/hyrax/my/works/_list_works.html.erb [hunk 2]
```diff
@@ -28,7 +28,15 @@
       </div>
     </div>
   </td>
-  <td class="date text-center"><%= document.date_modified %></td>
+  <td class="date text-center">
+    <% if document.date_modified.nil? %>
```
Comment (elrayle) [inline app/views/hyrax/my/works/_list_works.html.erb:32]: I think the reason something like this wasn't done before is that date_modified and date_uploaded are core fields and are set by the system, not the user.  I wonder if you can put just the change to this file in a separate PR.  Everything else looks good to go.  I'd like to think about this change.

**Review by elrayle — CHANGES_REQUESTED** [review]

**Comment by sdellis** [comment]
@elrayle I extracted the date formatting from this PR, so this is ready for final review. Will create a new PR for that one.

**Review by elrayle — APPROVED** [review]
Thanks for getting this in place.  These consistency updates to the UI will really improve the user experience.

**Pull request merged**